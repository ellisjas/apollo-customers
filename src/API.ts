/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type CreateZellerCustomersInput = {
  email: string,
  name: string,
<<<<<<< Updated upstream
  role: Role,
};

export enum Role {
  ADMIN = "ADMIN",
  MANAGER = "MANAGER",
}
=======
  role: string,
};
>>>>>>> Stashed changes


export type ZellerCustomers = {
  __typename: "ZellerCustomers",
  id: string,
  email: string,
  name: string,
<<<<<<< Updated upstream
  role: Role,
=======
  role: string,
>>>>>>> Stashed changes
};

export type UpdateZellerCustomersInput = {
  id: string,
  email?: string | null,
  name?: string | null,
<<<<<<< Updated upstream
  role?: Role | null,
=======
  role?: string | null,
>>>>>>> Stashed changes
};

export type DeleteZellerCustomersInput = {
  id: string,
};

export type TableZellerCustomersFilterInput = {
  id?: TableIDFilterInput | null,
  email?: TableStringFilterInput | null,
  name?: TableStringFilterInput | null,
  role?: TableStringFilterInput | null,
};

export type TableIDFilterInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
};

export type TableStringFilterInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
};

export type ZellerCustomersConnection = {
  __typename: "ZellerCustomersConnection",
  items?:  Array<ZellerCustomers | null > | null,
  nextToken?: string | null,
};

export type CreateZellerCustomersMutationVariables = {
  input: CreateZellerCustomersInput,
};

export type CreateZellerCustomersMutation = {
  createZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};

export type UpdateZellerCustomersMutationVariables = {
  input: UpdateZellerCustomersInput,
};

export type UpdateZellerCustomersMutation = {
  updateZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};

export type DeleteZellerCustomersMutationVariables = {
  input: DeleteZellerCustomersInput,
};

export type DeleteZellerCustomersMutation = {
  deleteZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};

export type GetZellerCustomersQueryVariables = {
  id: string,
};

export type GetZellerCustomersQuery = {
  getZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};

export type ListZellerCustomersQueryVariables = {
  filter?: TableZellerCustomersFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListZellerCustomersQuery = {
  listZellerCustomers?:  {
    __typename: "ZellerCustomersConnection",
    items?:  Array< {
      __typename: "ZellerCustomers",
      id: string,
      email: string,
      name: string,
<<<<<<< Updated upstream
      role: Role,
=======
      role: string,
>>>>>>> Stashed changes
    } | null > | null,
    nextToken?: string | null,
  } | null,
};

export type OnCreateZellerCustomersSubscriptionVariables = {
  id?: string | null,
  email?: string | null,
  name?: string | null,
<<<<<<< Updated upstream
  role?: Role | null,
=======
  role?: string | null,
>>>>>>> Stashed changes
};

export type OnCreateZellerCustomersSubscription = {
  onCreateZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};

export type OnUpdateZellerCustomersSubscriptionVariables = {
  id?: string | null,
  email?: string | null,
  name?: string | null,
<<<<<<< Updated upstream
  role?: Role | null,
=======
  role?: string | null,
>>>>>>> Stashed changes
};

export type OnUpdateZellerCustomersSubscription = {
  onUpdateZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};

export type OnDeleteZellerCustomersSubscriptionVariables = {
  id?: string | null,
  email?: string | null,
  name?: string | null,
<<<<<<< Updated upstream
  role?: Role | null,
=======
  role?: string | null,
>>>>>>> Stashed changes
};

export type OnDeleteZellerCustomersSubscription = {
  onDeleteZellerCustomers?:  {
    __typename: "ZellerCustomers",
    id: string,
    email: string,
    name: string,
<<<<<<< Updated upstream
    role: Role,
=======
    role: string,
>>>>>>> Stashed changes
  } | null,
};
