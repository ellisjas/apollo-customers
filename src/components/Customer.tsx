import React from 'react'
import styled from 'styled-components'
import { defaultFont, Heading3, Paragraph } from '../utils'
import { ZellerCustomer } from '../__generated__/graphql'

const Container = styled.div`
  margin: 25px 0;
  display: flex;
  gap: 20px;
  align-items: center;
`

const Avatar = styled.div`
  font-family: ${defaultFont};
  background-color: #ebf2fa;
  color: #619edd;
  border-radius: 5px;
  width: 40px;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
`
const Details = styled.div`
  display: flex;
  flex-direction: column;
  gap: 5px;
`

const Customer = ({ name, role }: ZellerCustomer) => {
  if (!name || !role) return <div />

  return (
    <Container>
      <Avatar>{name.charAt(0).toUpperCase()}</Avatar>
      <Details>
        <Heading3>{name}</Heading3>
        <Paragraph>{role.charAt(0) + role.slice(1).toLowerCase()}</Paragraph>
      </Details>
    </Container>
  )
}

export default Customer
