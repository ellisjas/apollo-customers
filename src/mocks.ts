import { ZellerCustomers } from "./API";

export const customer: ZellerCustomers = {
  id: '3457356745',
  email: 'jasmine@example.com',
  name: 'Jasmine',
  role: 'ADMIN',
  __typename: 'ZellerCustomers',
}

