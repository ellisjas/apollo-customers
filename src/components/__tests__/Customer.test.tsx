import React from 'react'
import Customer from '../Customer'
import { render } from '@testing-library/react'
import { ZellerCustomers } from '../../API'
import '@testing-library/jest-dom'
import { customer } from '../../mocks'

describe('components: <Customer />', () => {
  it('renders Customer', () => {
    const { container } = render(<Customer {...customer} />)
    expect(container).toMatchSnapshot()
  })
})
