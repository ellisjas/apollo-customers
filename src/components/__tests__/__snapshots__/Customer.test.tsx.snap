// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: <Customer /> renders Customer 1`] = `
<div>
  <div
    class="sc-hLBbgP gXRIlG"
  >
    <div
      class="sc-eDvSVe cslRUZ"
    >
      J
    </div>
    <div
      class="sc-jSUZER kalImh"
    >
      <h3
        class="sc-gswNZR dejOgB"
      >
        Jasmine
      </h3>
      <p
        class="sc-dkrFOg jINReh"
      >
        Admin
      </p>
    </div>
  </div>
</div>
`;
